pages/uat_actions/generate_test_run.vue

```diff
@@ -91,22 +125,36 @@ export default {
   },
   data() {
     return {
+      status: statuses.Idle,
       form: initialFormState(),
       addToDart: false,
       plateSpecs: [],
+      maxNumberOfPlates: MAX_NUMBER_OF_PLATES,
+      maxNumberOfPositives: MAX_NUMBER_OF_POSITIVES
     }
   },
   computed: {
     totalPlates () {
       return this.plateSpecs.reduce(function (acc, obj) { return acc + obj.numberOfPlates }, 0)
+    },
+    isBusy() {
+      return this.status === statuses.Busy
+    },
+    isValid() {
+      return this.totalPlates>0 && this.totalPlates<=200
+    },
+    totalPlatesSyle() {
+      return {
+          color: this.totalPlates === 200 ? 'red' : 'black'
+      }
     }
   },
   methods: {
     numberOfPositivesOptions() {
-      return Array.from({length: MAX_NUMBER_OF_POSITIVES + 1}, (v, i) => i) // 0 - 96
+      return Array.from({length: this.maxNumberOfPositives + 1}, (v, i) => i) // 0 - 96
     },
     numberOfPlatesOptions() {
-      return Array.from({length: MAX_NUMBER_OF_PLATES}, (v, i) => i + 1) // 1 - 200
+      return Array.from({length: this.maxNumberOfPlates - this.totalPlates}, (v, i) => i + 1) // 200 - totalPlates
```

Comment:
I'm intrigued what happens to the drop down when you add exactly 200 plates?  It becomes a drop down with no options in it?  I see you disable the "Add" button in this scenario, but maybe the whole add row should disappear when you hit 200?

Reply:
Yes, it had left an empty drop down so I had made the 200/200 turn red. But think your right - just added a v-if, to only show the drop down if total isn't reached 

![Screenshot 2021-07-21 at 15 16 54](https://user-images.githubusercontent.com/11446617/126503960-7925fecb-abe3-402d-b1cd-e25251f38df3.png)